### ISSUE TITLE
[FEATURE] Easily create a new todo / checklist item by pressing Enter before EOL

### ISSUE THREAD

**Issue by dandv** [pbek/QOwnNotes#3192]
#### Is your feature request related to a problem? Please describe.

When I edit a `* [ ]` TODO list, sometimes I want to easily add an item at the beginning of the list. 

#### Describe the solution you'd like

Pressing <kbd>Enter</kbd> after the first `*` in the list, or after the `* [ ]`, could create a new item.

#### Describe alternatives you've considered

I could of course navigate before the first item in the list and type <kbd>Enter</kbd>, <kbd>* [ ] first item</kbd>

#### Additional context

[Here's a screencast narrating what happens when I press Enter in those locations](https://github.com/user-attachments/assets/17ed29fd-3655-4041-9564-5fa80bf4caf8)

**Comment by pbek** [comment]
I'm not sure if I understand what you are talking about.
If you are pressing Enter at the EOL in a checkbox list, a new checkbox item is already created.

**Comment by dandv** [comment]
Right. The issue is, how do I easily insert a *first* item in the checkbox list? There's no previous checkbox item at the end of which to press Enter.

**Comment by pbek** [comment]
Create a 2nd item and press `Ctrl+Shift+Up` to move it up.

**Comment by dandv** [comment]
Nice shortcut, thank you for that workaround. Would there be any downsides to implementing the suggestion above, to create a blank first item when pressing Enter after the first `* ` or `* [ ]`?

Before, where `|` denotes caret position:
```
* [ ]| TODO item 2
* [ ] item 3
```

After pressing Enter at the caret position:
```
* [ ]
* [ ] TODO item 2
* [ ] item 3
```

If the downsides aren't worth the convenience, feel free to close the issue.

**Comment by pbek** [comment]
It's less about to do it than to break with existing behavior... 🤔 And if it would be intuitive. 
I'd also like to have a quick way to create a checkbox item but pressing Enter at each cursor position already does something...

> * [ ]| TODO item 2

Why would it be intuitive to insert a checkbox item above when pressing Enter at that position? 🤔
How would you get there quickly, so it even makes sense to do that if it would be intuitive?

**Comment by pbek** [comment]
I'd rather think about adding `- [ ] ` when you press Enter at the start of a checkbox line...

**Comment by pbek** [comment]
As a side note, you can also add a custom action (via scsripting engine) to do whatever you like with checkbox lists...

**Comment by pbek** [comment]
And just as another hint, calling a feature request "Minor feature" can be considered rude, unless you haven't already implemented it yourself and prepared a pull requests. 😉 In this case, it's also rather tricky, because there are so many edge cases that need to be considered...

**Comment by pbek** [comment]
Took the whole week to figure something out that was working and not breaking anything (at least I hope that's the case).

## 25.1.1
- If you press `Enter` at the start of the line of a checkbox list item or
  unordered list item, a new list item will be created above the current line
  (for [#3192](https://github.com/pbek/QOwnNotes/issues/3192))

**Comment by pbek** [comment]
There now is a new release, could you please test it and report if it works for you?
